Mathlib/Algebra/Homology/ShortComplex/RightHomology.lean

```diff
@@ -721,6 +721,177 @@ lemma opcyclesMap_comp [HasRightHomology S₁] [HasRightHomology S₂] [HasRight
+  rightHomologyMapIso' (Iso.refl _) _ _
+
+/-- The isomorphism `S.opcycles ≅ h.Q` induced by a right homology data `h` for a
+short complex `S`. -/
+noncomputable def opcyclesIso : S.opcycles ≅ h.Q :=
+  opcyclesMapIso' (Iso.refl _) _ _
+
+@[reassoc (attr := simp)]
+lemma p_comp_opcyclesIso_inv : h.p ≫ h.opcyclesIso.inv = S.pOpcycles := by
+  dsimp [pOpcycles, RightHomologyData.opcyclesIso]
+  simp only [p_opcyclesMap', id_τ₂, id_comp]
+
+@[reassoc (attr := simp)]
+lemma pOpcycles_comp_opcyclesIso_hom : S.pOpcycles ≫ h.opcyclesIso.hom = h.p := by
+  simp only [← h.p_comp_opcyclesIso_inv, assoc, Iso.inv_hom_id, comp_id]
+
+@[reassoc (attr := simp)]
+lemma rightHomologyIso_inv_comp_rightHomologyι :
+    h.rightHomologyIso.inv ≫ S.rightHomologyι = h.ι ≫ h.opcyclesIso.inv := by
+  dsimp only [rightHomologyι, rightHomologyIso, opcyclesIso, rightHomologyMapIso',
+    opcyclesMapIso', Iso.refl]
+  rw [rightHomologyι_naturality']
+
+@[reassoc (attr := simp)]
+lemma rightHomologyIso_hom_comp_ι :
+    h.rightHomologyIso.hom ≫ h.ι = S.rightHomologyι ≫ h.opcyclesIso.hom := by
+  simp only [← cancel_mono h.opcyclesIso.inv, ← cancel_epi h.rightHomologyIso.inv,
+    assoc, Iso.inv_hom_id_assoc, Iso.hom_inv_id, comp_id, rightHomologyIso_inv_comp_rightHomologyι]
+
+end RightHomologyData
+
+namespace RightHomologyMapData
+
+variable {φ : S₁ ⟶ S₂} {h₁ : S₁.RightHomologyData} {h₂ : S₂.RightHomologyData}
+  (γ : RightHomologyMapData φ h₁ h₂)
+
+lemma rightHomologyMap_eq [S₁.HasRightHomology] [S₂.HasRightHomology] :
+    rightHomologyMap φ = h₁.rightHomologyIso.hom ≫ γ.φH ≫ h₂.rightHomologyIso.inv := by
+  dsimp [RightHomologyData.rightHomologyIso, rightHomologyMapIso']
+  rw [← γ.rightHomologyMap'_eq, ← rightHomologyMap'_comp,
+    ← rightHomologyMap'_comp, id_comp, comp_id]
+  rfl
+
+lemma opcyclesMap_eq [S₁.HasRightHomology] [S₂.HasRightHomology] :
+    opcyclesMap φ = h₁.opcyclesIso.hom ≫ γ.φQ ≫ h₂.opcyclesIso.inv := by
+  dsimp [RightHomologyData.opcyclesIso, cyclesMapIso']
+  rw [← γ.opcyclesMap'_eq, ← opcyclesMap'_comp, ← opcyclesMap'_comp, id_comp, comp_id]
+  rfl
+
+lemma rightHomologyMap_comm [S₁.HasRightHomology] [S₂.HasRightHomology] :
+    rightHomologyMap φ ≫ h₂.rightHomologyIso.hom = h₁.rightHomologyIso.hom ≫ γ.φH := by
+  simp only [γ.rightHomologyMap_eq, assoc, Iso.inv_hom_id, comp_id]
+
+lemma opcyclesMap_comm [S₁.HasRightHomology] [S₂.HasRightHomology] :
+    opcyclesMap φ ≫ h₂.opcyclesIso.hom = h₁.opcyclesIso.hom ≫ γ.φQ := by
+  simp only [γ.opcyclesMap_eq, assoc, Iso.inv_hom_id, comp_id]
+
+end RightHomologyMapData
+
+section
+
+variable (C)
+variable [HasKernels C] [HasCokernels C]
+
+/-- The right homology functor `ShortComplex C ⥤ C`, where the right homology of a
+short complex `S` is understood as a kernel of the obvious map `S.fromOpcycles : S.opcycles ⟶ S.X₃`
+where `S.opcycles` is a cokernel of `S.f : S.X₁ ⟶ S.X₂`. -/
+@[simps]
+noncomputable def rightHomologyFunctor :
+    ShortComplex C ⥤ C where
```

Comment:
In the corresponding development of left homology, this is formatted as
```suggestion
noncomputable def rightHomologyFunctor : ShortComplex C ⥤ C where
```
It's fine to leave it like it is, of course, just something I noticed.